src/config.rs

```diff
@@ -0,0 +1,207 @@
+use std::env;
+use std::fs;
+use std::io::{Read, Write};
+use std::path::PathBuf;
+
+use ansi_term::{Colour, Style};
+use toml;
+use xdg::BaseDirectories;
+
+use error::TealdeerError::{self, ConfigError};
+
+pub const SYNTAX_CONFIG_FILE_NAME: &'static str = "syntax.toml";
+
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
+pub enum RawColour {
+    Black,
+    Red,
+    Green,
+    Yellow,
+    Blue,
+    Purple,
+    Cyan,
+    White,
+}
+
+impl From<RawColour> for Colour {
+    fn from(raw_colour: RawColour) -> Colour {
+        match raw_colour {
+            RawColour::Black => Colour::Black,
+            RawColour::Red => Colour::Red,
+            RawColour::Green => Colour::Green,
+            RawColour::Yellow => Colour::Yellow,
+            RawColour::Blue => Colour::Blue,
+            RawColour::Purple => Colour::Purple,
+            RawColour::Cyan => Colour::Cyan,
+            RawColour::White => Colour::White,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
+struct RawStyle {
+    pub foreground: Option<RawColour>,
+    pub background: Option<RawColour>,
+    pub underline: bool,
+    pub bold: bool,
+}
+
+impl Default for RawStyle {
+    fn default() -> RawStyle {
+        RawStyle{
+            foreground: None,
+            background: None,
+            underline: false,
+            bold: false,
+        }
+    }
+} // impl RawStyle
+
+impl From<RawStyle> for Style {
+    fn from(raw_style: RawStyle) -> Style {
+        let mut style = Style::default();
+
+        if let Some(foreground) = raw_style.foreground {
+            style = style.fg(Colour::from(foreground));
+        }
+
+        if let Some(background) = raw_style.background {
+            style = style.on(Colour::from(background));
+        }
+
+        if raw_style.underline {
+            style = style.underline();
+        }
+
+        if raw_style.bold {
+            style = style.bold();
+        }
+
+        style
+    }
+}
+
+#[derive(Debug, Default, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
+struct RawConfig {
+    pub highlight_style: RawStyle,
+    pub description_style: RawStyle,
+    pub example_text_style: RawStyle,
+    pub example_code_style: RawStyle,
+    pub example_variable_style: RawStyle,
+}
+
+impl RawConfig {
+    fn new() -> RawConfig {
+        let mut raw_config = RawConfig::default();
+
+        raw_config.highlight_style.foreground = Some(RawColour::Red);
+        raw_config.example_variable_style.underline = true;
```

Comment:
Not sure, what do we gain from having the caller specify them?

Reply:
When creating a new `RawConfig` instance, I would expect it to be uninitialized. Otherwise the responsibility of the `RawConfig` type (encapulating style configuration) and the calling program (configuring the style of the output) are mixed -- the container all of a sudden sets styling. I'd separate those clearly.

Reply:
That's what default does, if we move it to the caller site we'd repeat the same thing 3 times.

Reply:
Ok, let's keep it for now, but then at least add a docstring indicating that some values will be initialized. Something like this:

```
/// Create a new `RawConfig` instance with some values initialized to certain default values:
///
/// - Foreground highlight style is "red"
/// - Example variable style is "underlined"
```